.github/scripts/trymerge.py

```diff
@@ -1265,6 +1321,86 @@ def checks_to_str(checks: List[Tuple[str, Optional[str]]]) -> str:
 def checks_to_markdown_bullets(checks: List[Tuple[str, Optional[str]]]) -> List[str]:
     return [f"- [{c[0]}]({c[1]})" if c[1] is not None else f"- {c[0]}" for c in checks[:5]]
 
+def get_flaky_rules(url: str, num_retries: int = 3) -> List[FlakyRule]:
+    try:
+        return [FlakyRule(**rule) for rule in fetch_json_list(url)]
+    except Exception as e:
+        print(f"Could not download {url} because: {e}.")
+        if num_retries > 0:
+            return get_flaky_rules(url, num_retries=num_retries - 1)
+        return []
+
+def get_rockset_results(head_sha: str, merge_base: str, num_retries: int = 3) -> List[Dict[str, Any]]:
+    query = f"""
+SELECT
+    w.name as workflow_name,
+    j.id,
+    j.name,
+    j.conclusion,
+    j.completed_at,
+    j.html_url,
+    j.head_sha,
+    j.torchci_classification.captures as failure_captures,
+    LENGTH(j.steps) as steps,
+FROM
+    commons.workflow_job j join commons.workflow_run w on w.id = j.run_id
+where
+    j.head_sha in ('{head_sha}','{merge_base}')
+"""
+    try:
+        import rockset  # type: ignore[import]
+        res = rockset.RocksetClient(
+            host="api.usw2a1.rockset.com", api_key=os.environ["ROCKSET_API_KEY"]
+        ).sql(query)
+        return cast(List[Dict[str, Any]], res.results)
+    except Exception as e:
```

Comment:
Imo import error is not really retirable
```suggestion
    except ModuleNotFoundError:
      print("Could not use RockSet as rocket dependency is missing")
      return []
    except Exception as e:
```